### PR TITLE
add talk api client to get subject discussions

### PR DESCRIPTION
**Pull request by camallen**
Linked to https://github.com/zooniverse/Panoptes/pull/1833 

Add the panoptes talk client to find subjects with discussions. I'd like to refactor the panoptes client base functionality with composition but wanted to get a first pass by @marten for comments.

### CODE REVIEW

File: lib/panoptes/client/discussions.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      #
+      # @param focus_id [Integer] filter by focussable id
+      # @param focus_type [String] filter by focussable type
+      # @return list of discussions
+      def discussions(focus_id:, focus_type:)
+        query = {}
+        query[:focus_id] = focus_id
+        query[:focus_type] = focus_type
+
+        response = get("/discussions", query)
```
Comment (marten) [inline lib/panoptes/client/discussions.rb]: It's an authenticated call this way right? Will that return discussions for private projects?

Comment (marten) [inline lib/panoptes/client/discussions.rb]: Meant to say **un**authenticated

Comment (camallen) [inline lib/panoptes/client/discussions.rb]: figured - not sure at all to be honest.

**Comment by camallen** [comment]
@marten updated to add authenticated API req's for a talk client. Extracted the common client config / setup and added an `auth_url` param for talk login to different api url.

Lots of hound issues, happy to clean up.

**Pull request merged**